tests/app/views/test_buyers.py

```diff
@@ -1018,6 +1024,126 @@ def test_publish_button_unavailable_if_questions_not_answered(self, data_api_cli
+            ]
+        )
+
+        data_api_client.get_brief.return_value = api_stubs.brief(status="draft")
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists/requirements/"
+                              "digital-specialists/1234/publish")
+        page_html = res.get_data(as_text=True)
+
+        assert res.status_code == 200
+        assert 'This will show you what the supplier application deadline will be' in page_html
+        assert 'Your requirements will be open for' not in page_html
+
+    def test_correct_content_is_displayed_if_requirementLength_is_1_week(self, data_api_client):
+        self.login_as_buyer()
+        data_api_client.get_framework.return_value = api_stubs.framework(
+            slug='digital-outcomes-and-specialists',
+            status='live',
+            lots=[
+                api_stubs.lot(slug='digital-specialists', allows_brief=True)
+            ]
+        )
+
+        brief_json = api_stubs.brief(status="draft")
+        brief_questions = brief_json['briefs']
+        brief_questions.update({
+            'requirementsLength': '1 week'
+        })
+        data_api_client.get_brief.return_value = brief_json
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists/requirements/"
+                              "digital-specialists/1234/publish")
+        page_html = res.get_data(as_text=True)
+
+        assert res.status_code == 200
+        assert 'Your requirements will be open for 1 week.' in page_html
+        assert 'This will show you what the supplier application deadline will be' not in page_html
+        assert 'Your requirements will be open for 2 weeks' not in page_html
+
+    def test_correct_content_is_displayed_if_requirementLength_is_2_weeks(self, data_api_client):
+        self.login_as_buyer()
+        data_api_client.get_framework.return_value = api_stubs.framework(
+            slug='digital-outcomes-and-specialists',
+            status='live',
+            lots=[
+                api_stubs.lot(slug='digital-specialists', allows_brief=True)
+            ]
+        )
+
+        brief_json = api_stubs.brief(status="draft")
+        brief_questions = brief_json['briefs']
+        brief_questions.update({
+            'requirementsLength': '2 weeks'
+        })
+        data_api_client.get_brief.return_value = brief_json
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists/requirements/"
+                              "digital-specialists/1234/publish")
+        page_html = res.get_data(as_text=True)
+
+        assert res.status_code == 200
+        assert 'Your requirements will be open for 2 weeks.' in page_html
+        assert 'This will show you what the supplier application deadline will be' not in page_html
+        assert 'Your requirements will be open for 1 week' not in page_html
+
+    def test_correct_content_is_displayed_if_requirementLength_is_not_set(self, data_api_client):
```

Comment:
Yeah, this test should be `'requirementsLength': None` and not an empty string. It's the case when there is a `requirementsLength` key, but it has no value. This can come from the API. The other test has no `requirementsLength` key. Have updated the test, everything still passes.